### DISCUSSION TITLE
SSL Protocol Error

### DISCUSSION THREAD

**Discussion by Play-Tiger** [searxng/searxng#1444] [Q&A] [answered]
I just finished setting up searxng, but when I'm trying to enter the website it's just telling me this error "ERR_SSL_PROTOCOL_ERROR".
I've set up .env like this and I created an encryption key.
![Inked177201347-1c074a91-9fb7-4d6e-8101-2d6f19801ad3](https://user-images.githubusercontent.com/76755673/177210551-e976eee7-68b3-4914-b2b1-0cc0be10e416.jpg)
If you now look at the docker-compose.yaml file you can see that under environment the hostname and letsencrypt mail are still standard parameters.
![image](https://user-images.githubusercontent.com/76755673/177201571-d59a84fd-1b54-4adc-bfef-504ccb7ec169.png)

Does anybody know how to fix this?

**Comment by MrHappy** [comment]
Second screenshot looks fine. You should watch the environmentsetting of the container.
Can you access the site on the configured url?

**Comment by Play-Tiger** [comment]
No, I cant access the site with the url.

**Comment by MrHappy** [comment]
Can you ping it? What does the logging say? Etc etc? Some basic troubleshooting info would be helpfull...

**Comment by Play-Tiger** [comment]
I can ping the server. That works just fine. 
Where can I find the searxng logs?

**Comment by Play-Tiger** [comment]
If I do sudo docker-compose up I get these messages. It seems to me that they have something to do with this issue.

_Attaching to caddy, redis, searxng
caddy      | {"level":"info","ts":1656963735.7063706,"msg":"using provided configuration","config_file":"/etc/caddy/Caddyfile","config_adapter":"caddyfile"}
caddy      | {"level":"warn","ts":1656963735.7077587,"logger":"caddyfile","msg":"Unnecessary header_up X-Forwarded-Proto: the reverse proxy's default behavior is to pass headers to the upstream"}
caddy      | {"level":"warn","ts":1656963735.7138486,"msg":"Caddyfile input is not formatted; run the 'caddy fmt' command to fix inconsistencies","adapter":"caddyfile","file":"/etc/caddy/Caddyfile","line":2}
caddy      | {"level":"warn","ts":1656963735.7148163,"logger":"admin","msg":"admin endpoint disabled"}
caddy      | {"level":"info","ts":1656963735.7150843,"logger":"http","msg":"server is listening only on the HTTPS port but has no TLS connection policies; adding one to enable TLS","server_name":"srv0","https_port":443}
caddy      | {"level":"info","ts":1656963735.7151055,"logger":"http","msg":"enabling automatic HTTP->HTTPS redirects","server_name":"srv0"}
caddy      | {"level":"info","ts":1656963735.7155123,"logger":"tls.cache.maintenance","msg":"started background certificate maintenance","cache":"0xc0003fa2a0"}
caddy      | {"level":"info","ts":1656963735.7219186,"logger":"http","msg":"enabling automatic TLS certificate management","domains":["search.patechno.de"]}
caddy      | {"level":"info","ts":1656963735.7223744,"msg":"autosaved config (load with --resume flag)","file":"/config/caddy/autosave.json"}
caddy      | {"level":"info","ts":1656963735.7223873,"msg":"serving initial configuration"}
caddy      | {"level":"info","ts":1656963735.7246418,"logger":"tls.obtain","msg":"acquiring lock","identifier":"search.patechno.de"}
caddy      | {"level":"info","ts":1656963735.7257938,"logger":"tls.obtain","msg":"lock acquired","identifier":"search.patechno.de"}
caddy      | {"level":"info","ts":1656963735.7269034,"logger":"tls.issuance.acme","msg":"waiting on internal rate limiter","identifiers":["search.patechno.de"],"ca":"https://acme-v02.api.letsencrypt.org/directory","account":"myemail@patechno.de"}
caddy      | {"level":"info","ts":1656963735.7269201,"logger":"tls.issuance.acme","msg":"done waiting on internal rate limiter","identifiers":["search.patechno.de"],"ca":"https://acme-v02.api.letsencrypt.org/directory","account":"myemail@patechno.de"}
caddy      | {"level":"info","ts":1656963735.72761,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/data/caddy"}
caddy      | {"level":"info","ts":1656963735.7276385,"logger":"tls","msg":"finished cleaning storage units"}
caddy      | {"level":"info","ts":1656963736.9725423,"logger":"tls.issuance.acme.acme_client","msg":"trying to solve challenge","identifier":"search.patechno.de","challenge_type":"tls-alpn-01","ca":"https://acme-v02.api.letsencrypt.org/directory"}
redis      | 1:C 04 Jul 2022 19:42:17.711 # oO0OoO0OoO0Oo Redis is starting oO0OoO0OoO0Oo
redis      | 1:C 04 Jul 2022 19:42:17.711 # Redis version=7.0.2, bits=64, commit=00000000, modified=0, pid=1, just started
redis      | 1:C 04 Jul 2022 19:42:17.711 # Configuration loaded
redis      | 1:M 04 Jul 2022 19:42:17.712 * monotonic clock: POSIX clock_gettime
redis      | 1:M 04 Jul 2022 19:42:17.715 * Running mode=standalone, port=6379.
redis      | 1:M 04 Jul 2022 19:42:17.716 # Server initialized
redis      | 1:M 04 Jul 2022 19:42:17.716 * Ready to accept connections
caddy      | {"level":"error","ts":1656963738.4164212,"logger":"tls.issuance.acme.acme_client","msg":"challenge failed","identifier":"search.patechno.de","challenge_type":"tls-alpn-01","problem":{"type":"urn:ietf:params:acme:error:unauthorized","title":"","detail":"Cannot negotiate ALPN protocol \"acme-tls/1\" for tls-alpn-01 challenge","instance":"","subproblems":[]}}
caddy      | {"level":"error","ts":1656963738.4177465,"logger":"tls.issuance.acme.acme_client","msg":"validating authorization","identifier":"search.patechno.de","problem":{"type":"urn:ietf:params:acme:error:unauthorized","title":"","detail":"Cannot negotiate ALPN protocol \"acme-tls/1\" for tls-alpn-01 challenge","instance":"","subproblems":[]},"order":"https://acme-v02.api.letsencrypt.org/acme/order/617404016/103800169046","attempt":1,"max_attempts":3}
searxng    | SearXNG version 2022.07.04-b5cceeb2
searxng    | Use existing /etc/searxng/uwsgi.ini
searxng    | Use existing /etc/searxng/settings.yml
searxng    | Listen on 0.0.0.0:8080
searxng    | [uWSGI] getting INI configuration from /etc/searxng/uwsgi.ini
searxng    | [uwsgi-static] added mapping for /static => /usr/local/searxng/searx/static
searxng    | *** Starting uWSGI 2.0.20 (64bit) on [Mon Jul  4 19:42:18 2022] ***
searxng    | compiled with version: 11.2.1 20220219 on 21 December 2021 13:20:05
searxng    | os: Linux-5.4.0-121-generic #137-Ubuntu SMP Wed Jun 15 13:33:07 UTC 2022
searxng    | nodename: 9f88d524d046
searxng    | machine: x86_64
searxng    | clock source: unix
searxng    | pcre jit disabled
searxng    | detected number of CPU cores: 1
searxng    | current working directory: /usr/local/searxng
searxng    | detected binary path: /usr/sbin/uwsgi
searxng    | chdir() to /usr/local/searxng/searx/
searxng    | your memory page size is 4096 bytes
searxng    | detected max file descriptor number: 1048576
searxng    | building mime-types dictionary from file /etc/mime.types...1323 entry found
searxng    | lock engine: pthread robust mutexes
searxng    | thunder lock: disabled (you can enable it with --thunder-lock)
searxng    | *** Cache "searxngcache" initialized: 11MB (key: 2136 bytes, keys: 4272000 bytes, data: 8192000 bytes, bitmap: 250 bytes) preallocated ***
searxng    | uwsgi socket 0 bound to TCP address 0.0.0.0:8080 fd 4
searxng    | Python version: 3.10.4 (main, Apr 30 2022, 16:49:16) [GCC 11.2.1 20220219]
searxng    | Python main interpreter initialized at 0x7f4550f77b50
searxng    | python threads support enabled
searxng    | your server socket listen backlog is limited to 100 connections
searxng    | your mercy for graceful operations on workers is 60 seconds
searxng    | mapped 241344 bytes (235 KB) for 4 cores
searxng    | *** Operational MODE: threaded ***
searxng    | added /usr/local/searxng/ to pythonpath.
searxng    | spawned uWSGI master process (pid: 6)
searxng    | spawned uWSGI worker 1 (pid: 9, cores: 4)
searxng    | cache sweeper thread enabled
searxng    | spawned 1 offload threads for uWSGI worker 1
caddy      | {"level":"info","ts":1656963739.974967,"logger":"tls.issuance.acme.acme_client","msg":"trying to solve challenge","identifier":"search.patechno.de","challenge_type":"http-01","ca":"https://acme-v02.api.letsencrypt.org/directory"}
searxng    | 2022-07-04 19:42:20,762 ERROR:searx.shared: uwsgi.ini configuration error, add this line to your uwsgi.ini
searxng    | cache2 = name=searxcache,items=2000,blocks=2000,blocksize=4096,bitmap=1
searxng    | WSGI app 0 (mountpoint='') ready in 4 seconds on interpreter 0x7f4550f77b50 pid: 9 (default app)
caddy      | {"level":"error","ts":1656963743.012177,"logger":"tls.issuance.acme.acme_client","msg":"challenge failed","identifier":"search.patechno.de","challenge_type":"http-01","problem":{"type":"urn:ietf:params:acme:error:unauthorized","title":"","detail":"45.82.121.130: Invalid response from https://search.patechno.de/.well-known/acme-challenge/CU8ozjx3QyzrMbg6fL6Lgbhv9wgZsCBkCwCcAS3Yh_M: 404","instance":"","subproblems":[]}}
caddy      | {"level":"error","ts":1656963743.0134175,"logger":"tls.issuance.acme.acme_client","msg":"validating authorization","identifier":"search.patechno.de","problem":{"type":"urn:ietf:params:acme:error:unauthorized","title":"","detail":"45.82.121.130: Invalid response from https://search.patechno.de/.well-known/acme-challenge/CU8ozjx3QyzrMbg6fL6Lgbhv9wgZsCBkCwCcAS3Yh_M: 404","instance":"","subproblems":[]},"order":"https://acme-v02.api.letsencrypt.org/acme/order/617404016/103800178466","attempt":2,"max_attempts":3}
caddy      | {"level":"error","ts":1656963743.0139596,"logger":"tls.obtain","msg":"could not get certificate from issuer","identifier":"search.patechno.de","issuer":"acme-v02.api.letsencrypt.org-directory","error":"HTTP 403 urn:ietf:params:acme:error:unauthorized - 45.82.121.130: Invalid response from https://search.patechno.de/.well-known/acme-challenge/CU8ozjx3QyzrMbg6fL6Lgbhv9wgZsCBkCwCcAS3Yh_M: 404"}
caddy      | {"level":"info","ts":1656963743.0148623,"logger":"tls.issuance.acme","msg":"waiting on internal rate limiter","identifiers":["search.patechno.de"],"ca":"https://acme.zerossl.com/v2/DV90","account":"mymail@patechno.de"}
caddy      | {"level":"info","ts":1656963743.0152762,"logger":"tls.issuance.acme","msg":"done waiting on internal rate limiter","identifiers":["search.patechno.de"],"ca":"https://acme.zerossl.com/v2/DV90","account":"mymail@patechno.de"}_

**Reply by Play-Tiger** [reply]
The problem seems to be with caddy. It's constantly failing to encrypt.

**Reply by MrHappy** [reply]
I can't help you with caddy/letsencrypt.... I use haproxy and leysencrypt on a different server and removed caddy from this docker instance...

**Reply by Play-Tiger** [reply]
thanks for trying to help

**Comment by Play-Tiger** [comment, marked as answer]
I fixed the issue. The issue was my Domain. I bought a new domain and now it works perfectly fine!
